crates/typst-pdf/src/lib.rs

```diff
@@ -61,383 +74,482 @@ pub fn pdf(
     ident: Smart<&str>,
     timestamp: Option<Datetime>,
 ) -> Vec<u8> {
-    let mut ctx = PdfContext::new(document);
-    page::construct_pages(&mut ctx, &document.pages);
-    font::write_fonts(&mut ctx);
-    image::write_images(&mut ctx);
-    gradient::write_gradients(&mut ctx);
-    extg::write_external_graphics_states(&mut ctx);
-    pattern::write_patterns(&mut ctx);
-    write_named_destinations(&mut ctx);
-    page::write_page_tree(&mut ctx);
-    page::write_global_resources(&mut ctx);
-    write_catalog(&mut ctx, ident, timestamp);
-    ctx.pdf.finish()
-}
-
-/// Context for exporting a whole PDF document.
-struct PdfContext<'a> {
-    /// The document that we're currently exporting.
-    document: &'a Document,
-    /// The writer we are writing the PDF into.
+    PdfBuilder::new(document)
```

Comment:
```suggestion
    PdfBuilder::<BuildContent, Resources<()>>::new(document)
```
I think it's nice if we can see the initial state as well.

Reply:
We can't see the state anymore, since `transition` is no longer a thing. Should I make it explicit everywhere?